screens/signUp.js

```diff
@@ -1,11 +1,16 @@
+/* eslint-disable no-unused-vars */
+import { useState } from "react";
 import { Image, View, Text, TouchableOpacity } from "react-native";
 import { useTailwind } from "tailwind-rn";
 
 import { IconInput } from "../components/inputs";
-import { RoundButton } from "../components/buttons";
 
 const SignUp = () => {
   const tailwind = useTailwind();
+  const [email, setEmail] = useState("");
```

Comment:
It looks like you are using these states to save the input data the user inputs in these fields (email,  name,  password, dbInsert). However since you are not doing anything with that data until the user clicks the signup button, there is no need to keep saving the data every time the use makes a change to the field. Instead, we should use the `useRef` hook to get the value of the fields when the user clicks the Signup Button.

https://reactjs.org/docs/hooks-reference.html#useref

Reply:
I have tried using "useRef" but not sure it's the same way you asked for, have a look once you get chance.